src/date-input/date-input.tsx

```diff
@@ -369,6 +680,18 @@ export const DateInput = ({
                     maxLength={2}
                     value={monthValue}
                     ref={monthInputRef}
+                    {...(hoverValue !== undefined && {
```

Comment:
i find this hard to understand. Could you consider a helper function to do this instead? Then assign an undefined to the prop rather than conditionally adding.